ComponentKit/Core/CKComponentMemoizer.mm

```diff
@@ -149,6 +185,11 @@ id CKMemoize(CKMemoizationKey memoizationKey, id (^block)(void))
 {
   _CKComponentMemoizerImpl *impl = [_CKComponentMemoizerImpl currentMemoizer];
   CKComponent *component = [impl dequeueComponentForKey:memoizationKey];
+  if (component) {
+    if (!CKCanUseMemoizedComponent(component)) {
```

Comment:
Good question. I wrote some tests, so going to try changing it and re-running them. One thing is that we could probably leave the old component in the queue if it's not usable if I move the call into `dequeue`.